### DISCUSSION TITLE
Seeing some major FOUC with Solid-Start / Vite setup

### DISCUSSION THREAD

**Discussion by caseybaggz** [chakra-ui/panda#1259] [Q&A]
Hello! I've officially migrated from Tailwind to Panda in my Solid-Start project and am now seeing some pretty gnarly FOUC (about 2 seconds) that never existed until I migrated. I've tried a few different scenarios, all of which still do not fix the issue in production so I'm wondering if there is something I'm missing on the Panda side?

## Context:

- Setup via PostCSS/Solid recommendation
- Solid-Start is basically Remix for SolidJS, still in Beta but so far pretty stable

### 1. Using global css import = FOUC

When importing to `root.css` in the `root.tsx` file, results in FOUC

```tsx
// root.tsx - entry to app

import './root.css'
```

### 2. Preloading root CSS = FOUC

When attempting to preload the styles with a `Link` (basically Remix installation) also results in FOUC locally. 

🐛 There is a bug in the production version of this though from the Vite output not finding the root.css file - checking on how to fix that in the interim.

```tsx
// root.tsx - entry to app

import './root.css'

...

<Head>
  <Link rel="preload" as="style" href="../src/root.css" />
</Head>
```

### 3. Binding styles via the module = FOUC

When trying to just bind the module/raw output via the `Style` component, results in FOUC

```tsx
// root.tsx - entry to app

import styles from './root.css'

...

<Style>{styles}</Style>
```

Would love to know how to fix this so I can promote my app publicly soon. 👀  Thanks ahead of time for any support!

## Visual
https://github.com/chakra-ui/panda/assets/4819738/094f1b98-009d-4ddf-8939-2c497f9deee7

**Comment by anubra266** [comment]
Hi @caseybaggz we'd need a minimal reproduction to be able to confirm and debug this.

**Comment by segunadebayo** [comment]
This looks like an issue with either the SSR setup of solid start + Vite. We'll need a repro or preview link to inspect this further.

**Reply by caseybaggz** [reply]
Hang tight! Let me get something up and running for you! Thanks for the help!

**Comment by caseybaggz** [comment]
@anubra266 @segunadebayo I setup a repo but it works fine. However, a big difference with my project is that it is now huge with a fully custom Panda setup down to multi-theming, semantic tokens, multiple recipes...everything. Panda generates the code so I don't think that's the issue though.

What I *did* notice yesterday is that Panda is still loading tons of `.mjs` files in the prod build vs the atomic CSS of only what's being used on each page. It's literally loading and using the entire library - even recipes I'm not using. I think this might be the issue.

Is there a private thread we can start somewhere (like Discord) where I can send you the URL to the prod site? It's not ready for public consumption yet, but I think it will help you see what is happening on the Panda side?

Basically: Panda is using the JS files in production - not CSS. The only css being loaded despite using the `@layers...` is literally just the color tokens and reset. Everything else in pando is coming from `.mjs` files.

**Reply by caseybaggz** [reply]
I also want to note that I didn't have this behavior when using Tailwind or UnoCSS which I was under the impression Panda was influenced by and should work similarly in production build of the app?

**Reply by anubra266** [reply]
@caseybaggz My discord username is @anubra266. Please send a friend request, so I can access the site

**Reply by caseybaggz** [reply]
Continuing over at https://github.com/solidjs/solid-start/issues/1038
